library/network_connections.py

```diff
@@ -1034,7 +1044,11 @@ def connection_create(self, connections, idx, connection_current=None):
                     s_ip4.add_dns(nameserver["address"])
             for search_domain in ip["dns_search"]:
                 s_ip4.add_dns_search(search_domain)
-            s_ip4.clear_dns_options(True)
+            # NetworkManager ifcfg plguin will discard empty dns option which
+            # cause follow up NM.Connection.compare() raise false alarm
+            # Use False here to ask NetworkManager remove dns option completely instead
+            # of keeping an empty list
+            s_ip4.clear_dns_options(False)
```

Comment:
It seems that False is the default so maybe remove the argument and forget that it exists. I don't really understand the difference between True and False for this argument to be honest.

Reply:
True: use [] as dns option.
False: Remove the dns option entry from setting.